mlx/io/safetensors.cpp

```diff
@@ -120,34 +117,37 @@ SafetensorsLoad load_safetensors(
   // Load the json metadata
   auto rawJson = std::make_unique<char[]>(jsonHeaderLength);
   in_stream->read(rawJson.get(), jsonHeaderLength);
-  auto metadata = json::parse(rawJson.get(), rawJson.get() + jsonHeaderLength);
+  // TODO: remove this copy
+  std::string json_str(rawJson.get(), rawJson.get() + jsonHeaderLength);
+  auto metadata = io::parse_json(json_str);
   // Should always be an object on the top-level
-  if (!metadata.is_object()) {
+  if (!metadata.is<io::json_object>()) {
     throw std::runtime_error(
         "[load_safetensors] Invalid json metadata " + in_stream->label());
   }
   size_t offset = jsonHeaderLength + 8;
   // Load the arrays using metadata
   std::unordered_map<std::string, array> res;
   std::unordered_map<std::string, std::string> metadata_map;
-  for (const auto& item : metadata.items()) {
-    if (item.key() == "__metadata__") {
-      for (const auto& meta_item : item.value().items()) {
-        metadata_map.insert({meta_item.key(), meta_item.value()});
+  for (const auto& [item_key, item_value] : metadata.items()) {
+    if (item_key == "__metadata__") {
+      for (const auto& [meta_key, meta_value] : item_value.items()) {
+        metadata_map.insert({meta_key, meta_value});
       }
       continue;
     }
-    const std::string& dtype = item.value().at("dtype");
-    const Shape& shape = item.value().at("shape");
-    const std::vector<size_t>& data_offsets = item.value().at("data_offsets");
+    const std::string& dtype = item_value["dtype"];
+    // TODO: revert this to Shape
+    const std::vector<int>& shape = item_value["shape"];
```

Comment:
That's interesting, `Shape` is just an alias for `vector<int>`. It doesn't compile if you use it there?

Reply:
This was a red herring in the end. Shape works fine.